scifight/admin.py

```diff
@@ -124,6 +131,17 @@ def get_queryset(self, request):
         qs = super(FightStageAdmin, self).get_queryset(request)
         return qs.select_related('fight')
 
+    def formfield_for_foreignkey(self, db_field, request=None, **kwargs):
+        if db_field.name in self._fields_of_filter:
+            user = request.user
+            if not user.is_superuser:
+                if hasattr(user, 'scifight_user_profile'):
+                    tournament_of_user = user.scifight_user_profile.tournament
+                    kwargs["queryset"] = self._fields_of_filter[db_field.name].filter(
+                        tournament=tournament_of_user)
+
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
```

Comment:
In fact, we can. This field is present in every data model which is tournament specific, and we can require each subclass to additionally declare their `fields_of_filter` list.